packages/truffle-contract/lib/execute.js

```diff
@@ -31,38 +30,42 @@ var execute = {
           var bestEstimate = Math.floor(constructor.gasMultiplier * gas);
 
           // Don't go over blockLimit
-          (bestEstimate >= blockLimit)
+          bestEstimate >= blockLimit
             ? accept(blockLimit - 1)
             : accept(bestEstimate);
 
-        // We need to let txs that revert through.
-        // Often that's exactly what you are testing.
-        }).catch(err => accept());
+          // We need to let txs that revert through.
+          // Often that's exactly what you are testing.
+        })
+        .catch(() => accept());
     });
   },
 
   /**
    * Prepares simple wrapped calls by checking network and organizing the method inputs into
    * objects web3 can consume.
    * @param  {Object} constructor   TruffleContract constructor
+   * @param  {Object} methodABI     Function ABI segment w/ inputs & outputs keys.
    * @param  {Array}  _arguments    Arguments passed to method invocation
    * @return {Promise}              Resolves object w/ tx params disambiguated from arguments
    */
-  prepareCall: function(constructor, _arguments){
+  prepareCall: function(constructor, methodABI, _arguments) {
```

Comment:
I now pass the `methodABI` through so that it can be used in `utils.getTxParams`